plugins/module_utils/foreman_helper.py

```diff
@@ -380,7 +380,11 @@ def find_resource(self, resource, search, name=None, params=None, failsafe=False
         elif failsafe:
             result = None
         else:
-            self.fail_json(msg="No data found for %s" % search)
+            if len(results) > 1:
+                error_msg = "too many"
```

Comment:
Why not give the number in the error?
It could be a good hint while debugging whether it's 2 or 203410.

Reply:
```suggestion
                error_msg = "too many ({0})".format(len(results))
```

like this?